bundles/org.openhab.binding.playstation/src/main/java/org/openhab/binding/playstation/internal/discovery/PlayStationDiscovery.java

```diff
@@ -0,0 +1,508 @@
+     * host-request-port:997
+     * device-discovery-protocol-version:00020020
+     * system-version:07020001
+     * running-app-name:Youtube
+     * running-app-titleid:CUSA01116
+     *
+     * @param packet
+     * @return
+     */
+    private boolean parsePS4Packet(DatagramPacket packet) {
+        byte[] data = packet.getData();
+        String message = new String(data, StandardCharsets.UTF_8);
+        logger.debug("PS4 data '{}', length:{}", message, packet.getLength());
+
+        String ipAddress = packet.getAddress().toString().split("/")[1];
+        String hostId = "";
+        String hostType = "";
+        String hostName = "";
+        String hostPort = "";
+        String protocolVersion = "";
+        String systemVersion = "";
+
+        String[] rowStrings = message.trim().split("\\r?\\n");
+        for (String row : rowStrings) {
+            int index = row.indexOf(':');
+            index = index != -1 ? index : 0;
+            String key = row.substring(0, index);
+            String value = row.substring(index + 1);
+            switch (key) {
+                case RESPONSE_HOST_ID:
+                    hostId = value;
+                    break;
+                case RESPONSE_HOST_TYPE:
+                    hostType = value;
+                    break;
+                case RESPONSE_HOST_NAME:
+                    hostName = value;
+                    break;
+                case RESPONSE_HOST_REQUEST_PORT:
+                    hostPort = value;
+                    break;
+                case RESPONSE_DEVICE_DISCOVERY_PROTOCOL_VERSION:
+                    protocolVersion = value;
+                    if (!"00020020".equals(protocolVersion)) {
+                        logger.debug("Different protocol version: '{}'", protocolVersion);
+                    }
+                    break;
+                case RESPONSE_SYSTEM_VERSION:
+                    systemVersion = value;
+                    break;
+                default:
+                    break;
+            }
+        }
+        String hwVersion = hwVersionFromHostId(hostId);
+        String modelID = modelNameFromHostTypeAndHWVersion(hostType, hwVersion);
+        Map<String, Object> properties = new HashMap<>();
+        properties.put(IP_ADDRESS, ipAddress);
+        properties.put(IP_PORT, Integer.valueOf(hostPort));
+        properties.put(Thing.PROPERTY_MODEL_ID, modelID);
+        properties.put(Thing.PROPERTY_HARDWARE_VERSION, hwVersion);
+        properties.put(Thing.PROPERTY_FIRMWARE_VERSION, formatPS4Version(systemVersion));
+        properties.put(Thing.PROPERTY_MAC_ADDRESS, hostIdToMacAddress(hostId));
+        ThingUID uid = hostType.equalsIgnoreCase("PS5") ? new ThingUID(THING_TYPE_PS5, hostId)
+                : new ThingUID(THING_TYPE_PS4, hostId);
+
+        DiscoveryResult result = DiscoveryResultBuilder.create(uid).withProperties(properties).withLabel(hostName)
+                .withRepresentationProperty(hostId).build();
```

Comment:
The representation property must be the *name* of the property, not the value. Same for below.